File: Java/src/main/java/parrot/AfricanParrot.java
package parrot;

public class AfricanParrot extends Parrot{
    public AfricanParrot(ParrotTypeEnum type, int numberOfCoconuts, double voltage, boolean isNailed) {
        super(type, numberOfCoconuts, voltage, isNailed);
    }

    public AfricanParrot(int numberOfCoconuts, double voltage, boolean isNailed) {
        super(null, numberOfCoconuts, voltage, isNailed);
    }

    @Override
    public double getSpeed() {
        return Math.max(0, getBaseSpeed() - getLoadFactor() * numberOfCoconuts);
    }
}

File: Kotlin/src/main/kotlin/parrot/AfricanParrot.kt
package parrot

import kotlin.math.max

class AfricanParrot(numberOfCoconuts: Int, voltage: Double, isNailed: Boolean) :
        Parrot(numberOfCoconuts, voltage, isNailed){

            override val speed: Double
                get() = max(0.0, baseSpeed - loadFactor * numberOfCoconuts)
}